doc/ug/dv_methodology/index.md

```diff
@@ -299,60 +299,115 @@ Once the regression tool is developed, it will provide a way to enable these cap
-  These are usually coded in the testbench to check that the desired stimulus was generated properly, but can also be embedded in the RTL.
+* **Covergroup Coverage**:Observes values on buses, registers and so on.
+  This can verify that a specific value or range of values was seen on a bus, And that no illegal values were seen.
+  Simulators will throw an error if an illegal bin is filled.
+  Could also be the intersection of two cover points.
+  One example could be observing the FIFO full signal and the FIFO write enable signal is asserted at the same time.
+  This is called a coverage cross and will give valuable information on whether this scenario was ever seen. 
+* **Cover Property coverage**: This observes events or series of events.
+  As an example think of the TL UL register bus used.
+  A cover property cover point could be the handshaking between valid and ready.
+  There is a big overlap in what can be covered by SVA and Cover properties.
+  Suggestions on when to use which form of coverage is presented later in this document.
+
+
+#### Do we need both types of coverage
+
+Reaching a 100% code coverage does not tell you the whole story.
+Think of the simple example of an AND gate. 
+Input A, B, and output X. to get to a 100 percent code coverage only two different input combinations are needed.
+00 and 11, these two will produce all possible outcomes of X.
+
+![single AND gate](dv_method_single_gate.svg)
+
+
+The coverage will indicate that all code was exercised.
+But We do not know that our design work as intended. 
+All we know is that A, B, and X have been observed to take on both logic  0 and 1.
+We could not say for certain that the design was in fact an AND gate could just as easily be an OR gate.
+So we need functional coverage to tell us this.
+The first thing functional coverage will tell us is if we observed all possible values on the inputs.
+And by Adding a cross between the inputs and the output it will tell us which gate we are looking at.
+
+
+Reaching 100 percent functional coverage is not enough either.
+Remember functional coverage requires the designer to manually add coverage point into the design.
+Going back to our AND gate, let us say we take two of these and OR the outputs of the two.
+
+![multi gate](dv_method_multi_gate.svg)
+
 
-Code coverage can reach 100% but it may not be sufficient to indicate whether all interesting combinations of scenarios were exercised (such as boundary conditions, concurrent scenarios and different CSR fields holding specific combinations of values resulting in the DUT being exercised in interesting ways).
-Hence, developing a detailed functional coverage model in conjunction with constrained random test sequences is key to high quality DV effort.
+If we only add the cover points from our original example.
+These will still exercise the new output of our system and therefore result in reaching 100 percent functional coverage.
+But half of the design was not exercised.
```

Comment:
I would suggest combining these three lines into a single sentence. (The first two need combining, and I think all three works better)

Reply:
I agree